ast/expressions.go

```diff
@@ -238,7 +256,30 @@ type CaseExpr struct {
 
 // Restore implements Node interface.
 func (n *CaseExpr) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+	ctx.WriteKeyWord("CASE ")
+	if n.Value != nil {
+		if err := n.Value.Restore(ctx); err != nil {
+			return errors.Annotate(err, "An error occurred while restore CaseExpr.Value")
+		}
+		ctx.WritePlain(" ")
+	}
+	for idx, clause := range n.WhenClauses {
+		if idx != 0 {
```

Comment:
I think here the `if` condition can be avoided by changing the position of some spaces. For example, change `"CASE "` to `"CASE“`, change `"WHEN "` of `WhenClause` to `" WHEN "` (it doesn't matter since `WhenClause` only appears in `CaseExpr`), and move the above `ctx.WritePlain(" ")` before the `if`.
This is not a big case anyway, but you can fix the bug of `CaseExpr.Format` :-)